libc/stdio/vcscanf.c

```diff
@@ -34,6 +35,21 @@
     c;                       \
   })
 
+#define BUFFER_GROW 48
+#define BUFFER                                         \
+  ({                                                   \
+    int c = READ;                                      \
+    if (bufcur >= bufsize - 1) {                       \
+      bufsize = bufsize + BUFFER_GROW;                 \
+      buf = realloc(buf, bufsize);                     \
```

Comment:
Do we need dynamic memory if we're calling strtod()? Would it be possible to simply save a pointer to first character of the number, and then pass that to strtod()?

Reply:
Probably! I'd really prefer to avoid memory allocation if at all possible.

I chose this approach because of the callback interface which provides one character at a time. Is there a way to get a pointer to the string being scanned?

Reply:
I'm sorry I take this back. I forgot that this function has a character reader function. Sorry!

Reply:
No problem. I'm wondering if there's a better way to do this too. The parsing logic got somewhat complex due to the single character nature of the function.